ROOTFS/etc/waggle/sanity/fatal/check_hanwha_camera.test

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+
+import sys
+import os
+import json
+import urllib.error, urllib.request
+
+# TODO: credential denial by Hanwha camera is not captured
+#       by the http request; needs to look into the body
+#       to see if the response is 'unauthorized'
+
+"""
+    Error codes (1 byte)
+    each value represents the bit order
+"""
+NO_NODE_MANIFEST = 0
+READ_ERROR_MANIFEST = 1
+CAMERA_NOT_FOUND = 2
+FAILED_TO_GET_SNAPSHOT = 3
+UNLISTED_CAMERA = 4
+RESERVED = 5
+RESERVED2 = 6
+RESERVED3 = 7
+
+def print_log(msg, is_pass, title='Hanwha camera test'):
+    if is_pass:
+        print(f'{title}: {msg} PASS')
+    else:
+        print(f'{title}: {msg} FAIL')
+
+
+def read_camera_manifest(topic_camera='cameras', path='/etc/waggle/node_manifest.json'):
+    if not os.path.exists(path):
+        raise FileNotFoundError(f'manifest does not exist: {path}')
+
+    with open(path, 'r') as file:
+        try:
+            manifest = json.loads(file.read())
+        except:
+            raise IOError(f'Failed to load the manifest: {path}')
+        if topic_camera not in manifest:
+            raise EOFError(f'no cameras listed in the manifest, skipping test')
+        else:
+            for camera in manifest[topic_camera]:
+                if camera['present'] == True:
+                    yield camera
+
+
+def http_request(url, user='waggle', pw='a2kWe89wak3na7@', timeout=5):
+    passman = urllib.request.HTTPPasswordMgrWithDefaultRealm()
+    passman.add_password(None, url, user, pw)
+    authhandler = urllib.request.HTTPDigestAuthHandler(passman)
+    opener = urllib.request.build_opener(authhandler)
+    urllib.request.install_opener(opener)
+    headers = {'Accept': 'application/json'}
+    req = urllib.request.Request(url, headers=headers)
+    try:
+        with urllib.request.urlopen(req, timeout=timeout) as response:
+            return response.code, response.read()
+    except urllib.error.URLError:
+        return 404, b''
+
+
+def check(camera_ip_domain='10.31.81.'):
+    try:
+        cameras = list(read_camera_manifest())
+    except FileNotFoundError as ex:  # failed to locate the manifest
+        print_log(str(ex), is_pass=False)
+        return (1 << NO_NODE_MANIFEST)
+    except IOError as ex:  # failed to read the manifest
+        print_log(str(ex), is_pass=False)
+        return (1 << READ_ERROR_MANIFEST)
+    except EOFError as ex:  # if no cameras provided, no need to test
```

Comment:
I'm surprised by this one. I would have thought you only return this if you cleanly read cameras and check for:
```
if len(cameras) == 0:
  return 0
```

Reply:
You are right. Logic error. I changed it. Thanks!